src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1555,7 +1555,7 @@ public IProgressManager getProgressManager()
     public void addVisitingPlayer(final PlayerEntity player)
     {
         final Rank rank = getPermissions().getRank(player);
-        if (rank != Rank.OWNER && rank != Rank.OFFICER && !visitingPlayers.contains(player) && MineColonies.getConfig().getServer().sendEnteringLeavingMessages.get())
+        if (rank != getPermissions().getRankOwner() && rank != getPermissions().getRankOfficer() && !visitingPlayers.contains(player) && MineColonies.getConfig().getServer().sendEnteringLeavingMessages.get())
         {
```

Comment:
I believe we need a new permission flag for this now. Since "co-owner" or similarly assigned ranks from the player should also not get this.